hudi-client/src/main/java/org/apache/hudi/execution/CopyOnWriteLazyInsertIterable.java

```diff
@@ -50,15 +51,23 @@
   protected final HoodieTable<T> hoodieTable;
   protected final String idPrefix;
   protected int numFilesWritten;
+  protected Supplier<Integer> idSupplier;
+  protected Supplier<Integer> stageSupplier;
+  protected Supplier<Long> attemptSupplier;
 
   public CopyOnWriteLazyInsertIterable(Iterator<HoodieRecord<T>> sortedRecordItr, HoodieWriteConfig config,
-                                       String instantTime, HoodieTable<T> hoodieTable, String idPrefix) {
+                                       String instantTime, HoodieTable<T> hoodieTable, String idPrefix,
+                                       Supplier<Integer> idSupplier, Supplier<Integer> stageSupplier,
+                                       Supplier<Long> attemptSupplier) {
```

Comment:
Actually, I am not sure if we can package these three args into a DTO structure. Just a thought, you can ignore.

Reply:
> Actually, I am not sure if we can package these three args into a DTO structure. Just a thought, you can ignore.

Yes, I think it is better.